File: 2020/day01/Charly98cma/day01.py
l = open("input.txt", "r").read().split('\n')[:-1]
found = False
for x in l:
    for y in l:
        if (int(x)+int(y)==2020):
            print("1st STAR SOLUTION ->", int(x)*int(y))
            found = True
            break
    if (found):
        break

found = False
for x in l:
    for y in l:
        for z in l:
            if (int(x)+int(y)+int(z)==2020):
                print("2nd STAR SOLUTION ->", int(x)*int(y)*int(z))
                found = True
                break
        if (found):
            break
    if (found):
        break

File: 2022/day01/jfagoagas/day01.go
package main

import (
	"bufio"
	"flag"
	"fmt"
	"os"
	"sort"
	"strconv"
)

var (
	inputFile = flag.String("f", "input.txt", "Puzzle input file")
)

func exitOnError(err error) {
	if err != nil {
		fmt.Println(err)
		os.Exit(1)
	}
}

func readLines(path string) []int {
	file, err := os.Open(path)
	exitOnError(err)
	defer func() {
		err := file.Close()
		exitOnError(err)
	}()

	scanner := bufio.NewScanner(file)

	var lines []int
	for scanner.Scan() {
		if scanner.Text() == "" {
			lines = append(lines, 0)
			continue
		}
		line, _ := strconv.Atoi(scanner.Text())
		lines = append(lines, line)

	}
	return lines
}

func countCalories(cals []int) []int {
	var totalCalories []int
	currentCalories := 0
	for index, cal := range cals {
		currentCalories += cal
		// New Elf or last
		if cal == 0 || index == len(cals)-1 {
			totalCalories = append(totalCalories, currentCalories)
			currentCalories = 0
		}

	}
	sort.Ints(totalCalories)
	return totalCalories
}

func main() {
	// Parse Flags
	flag.Parse()
	// Read Input file
	input := readLines(*inputFile)

	totalCalories := countCalories(input)
	fmt.Printf("Part 1 - Result: %d\n", totalCalories[len(totalCalories)-1])
	top3Calories := totalCalories[len(totalCalories)-1] + totalCalories[len(totalCalories)-2] + totalCalories[len(totalCalories)-3]
	fmt.Printf("Part 2 - Result: %d\n", top3Calories)
}
